### DISCUSSION TITLE
v3.0.0

### DISCUSSION THREAD

**Discussion by AykutSarac** [AykutSarac/jsoncrack.com#336] [Announcements]
# JSON Crack - v3

## What has changed since version 2?

* New UI
* Edit nodes on graph
* Display path to nodes at Node Modal
* JWT Decoder
* JSON Schema
* Live/Manual Transform toggle
* Performance improvements
* New Landing Page
* Account, Cloud functions, save/share documents
* Fetch JSON through URL
* Image Previews at graph
* Collapse/Expand toggle all nodes
* Display children count on parent nodes

Kudos to @AykutSarac @hanbin9775 @nnamua @liqun98 @roland-becsi-veratrak @cihat @mbrukman @yuvalg99 @Hexrt @dogukanuhn @VladCuciureanu  @shivam1646

<img width="1440" alt="image" src="https://github.com/AykutSarac/jsoncrack.com/assets/47941171/c9743625-c31b-4fb0-8c94-ec5544cc6542">

## Changelog
* fix hide wrong collapsed nodes by @victorbrambati in https://github.com/AykutSarac/jsoncrack.com/pull/107
* jsoncrack.com Migration by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/113
* Add mobile support for JSON Crack by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/115
* Enable Performance Mode by default by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/132
* General UI/UX Improvements by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/133
* Link Travis CI badge to project build status by @mbrukman in https://github.com/AykutSarac/jsoncrack.com/pull/109
* Update Docker section formatting in README by @mbrukman in https://github.com/AykutSarac/jsoncrack.com/pull/108
* Bug/share link by @yuvalg99 in https://github.com/AykutSarac/jsoncrack.com/pull/136
* Improve Editor Rerender Performance by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/157
* Prevent All Nodes from Expanding by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/167
* fix: highlight red-dashed border display by @Hexrt in https://github.com/AykutSarac/jsoncrack.com/pull/165
* Handle window freeze by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/191
* Canvas should fit into view on click Center icon at toolbar by @dogukanuhn in https://github.com/AykutSarac/jsoncrack.com/pull/203
* Accept JSONC by stripping it of it's comments by @VladCuciureanu in https://github.com/AykutSarac/jsoncrack.com/pull/227
* Added children count label next to the node key by @VladCuciureanu in https://github.com/AykutSarac/jsoncrack.com/pull/218
* Added a 'Collapse/Expand Graph' button by @VladCuciureanu in https://github.com/AykutSarac/jsoncrack.com/pull/230
* Fix: persist hidden nodes on rotate by @shivam1646 in https://github.com/AykutSarac/jsoncrack.com/pull/238
* Fix: don't show disconnected nodes on expand by @shivam1646 in https://github.com/AykutSarac/jsoncrack.com/pull/243
* improve language of "huge file" error by @TallTed in https://github.com/AykutSarac/jsoncrack.com/pull/210
* fix: #158 by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/250
* chore: code refactor by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/252
* fix empty nodes by @victorbrambati in https://github.com/AykutSarac/jsoncrack.com/pull/251
* seperate colors from array and object by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/264
* Cloud updates by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/280
* Bump loader-utils from 2.0.2 to 2.0.4 by @dependabot in https://github.com/AykutSarac/jsoncrack.com/pull/281
* fix: align object's children to keep the consistency of graph illustration by @hanbin9775 in https://github.com/AykutSarac/jsoncrack.com/pull/268
* Bump json5 from 1.0.1 to 1.0.2 by @dependabot in https://github.com/AykutSarac/jsoncrack.com/pull/299
* Fixed README typo by @nnamua in https://github.com/AykutSarac/jsoncrack.com/pull/295
* import file by drag and drop by @liqun98 in https://github.com/AykutSarac/jsoncrack.com/pull/284
* Fix typo by @hslee2008 in https://github.com/AykutSarac/jsoncrack.com/pull/302
* Fix smooth scroll behavior on styles by @cihat in https://github.com/AykutSarac/jsoncrack.com/pull/291
* Dogukanuhn/main by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/309
* #257 Getting path of JSON on clicking the edges by @hanbin9775 in https://github.com/AykutSarac/jsoncrack.com/pull/304
* Update default.conf by @roland-becsi-veratrak in https://github.com/AykutSarac/jsoncrack.com/pull/274
* Feat/mantine UI by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/318
* Refactor codebase by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/326
* Version 3 by @AykutSarac in https://github.com/AykutSarac/jsoncrack.com/pull/335

## New Contributors
* @mbrukman made their first contribution in https://github.com/AykutSarac/jsoncrack.com/pull/109
* @yuvalg99 made their first contribution in https://github.com/AykutSarac/jsoncrack.com/pull/136
* @Hexrt made their first contribution in https://github.com/AykutSarac/jsoncrack.com/pull/165
* @dogukanuhn made their first contribution in https://github.com/AykutSarac/jsoncrack.com/pull/203
* @VladCuciureanu made their first contribution in https://github.com/AykutSarac/jsoncrack.com/pull/227
* @shivam1646 made their first contribution in https://github.com/AykutSarac/jsoncrack.com/pull/238
* @TallTed made their first contribution in https://github.com/AykutSarac/jsoncrack.com/pull/210
* @hanbin9775 made their first contribution in https://github.com/AykutSarac/jsoncrack.com/pull/268
* @nnamua made their first contribution in https://github.com/AykutSarac/jsoncrack.com/pull/295
* @liqun98 made their first contribution in https://github.com/AykutSarac/jsoncrack.com/pull/284
* @roland-becsi-veratrak made their first contribution in https://github.com/AykutSarac/jsoncrack.com/pull/274

**Full Changelog**: https://github.com/AykutSarac/jsoncrack.com/compare/v2.0.0...v3.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/AykutSarac/jsoncrack.com/releases/tag/v3.0.0'>v3.0.0</a>.</em>

**Comment by jeremyfiel** [comment]
can you clarify what support of "JSON Schema" was added in v3?

**Reply by AykutSarac** [reply]
JSON Crack uses Monaco Editor by Microsoft which supports all draft versions from Draft 4 to JSON Schema Draft 2020-12.
